ocaml/tests/test_event.ml

```diff
@@ -57,6 +57,36 @@ let test_event_from_ev_rel () =
   Alcotest.(check int) "list length" 1 (List.length vm_ev);
   Alcotest.(check int) "list length" 1 (List.length vbd_ev)
 
+let test_events_from_obj () =
+  (* Test that events on one object create the correct number of events on other objects *)
+  let __context, session_id = event_setup_common () in
+  let evs = Xapi_event.from __context ["vm"] "" 30.0 |> parse_event_from in
+  let vma = make_vm __context () in
+  let tok = evs.token in
+  let vm_ev = List.filter (fun ev -> ev.ty="vm") evs.events in
+  Alcotest.(check int) "list length" 1 (List.length vm_ev);
+
+  let evs2 = Xapi_event.from __context ["vm"] tok 30.0 |> parse_event_from in
+  let vmb = make_vm __context () in
+  let tok2 = evs2.token in
+  let vm_ev2 = List.filter (fun ev -> ev.ty="vm") evs2.events in
+  Alcotest.(check int) "list length" 1 (List.length vm_ev2);
+
+  let evs3 = Xapi_event.from __context ["vm"] tok2 30.0 |> parse_event_from in
+  let vbd = make_vbd ~__context ~vM:vma () in
+  let tok3 = evs3.token in
+  let vm_ev3 = List.filter (fun ev -> ev.ty="vm") evs3.events in
+  Alcotest.(check int) "list length" 1 (List.length vm_ev3);
+  let vbd_ev = List.filter (fun ev -> ev.ty="vbd") evs3.events in
+  Alcotest.(check int) "list length" 0 (List.length vbd_ev);
+
+  let evs4 = Xapi_event.from __context ["vm"] tok3 30.0 |> parse_event_from in
+  Db.VBD.set_VM __context vbd vmb;
```

Comment:
The only thing you've changed in the events code in this PR is what happens after a WriteField event, and you're not actually invoking the events logic after this, which is the only thing that can cause a WriteField event.